src/core/DrumSynth.cpp

```diff
@@ -172,34 +171,26 @@ void DrumSynth::GetEnv(int env, const char* sec, const char* key, QString ini)
 
 float DrumSynth::waveform(float ph, int form)
 {
-	float w;
+	float w, p = ediv(ph, numbers::tau_v<float>);
 
 	switch (form)
 	{
-	case 0:
-		w = static_cast<float>(sin(fmod(ph, TwoPi)));
-		break; // sine
-	case 1:
-		w = static_cast<float>(fabs(2.0f * static_cast<float>(sin(fmod(0.5f * ph, TwoPi))) - 1.f));
-		break; // sine^2
-	case 2:
-		while (ph < TwoPi)
-		{
-			ph += TwoPi;
-		}
-		w = 0.6366197f * static_cast<float>(fmod(ph, TwoPi) - 1.f); // tri
-		if (w > 1.f)
-		{
-			w = 2.f - w;
-		}
+	case 0: // sine
+		w = std::sin(p);
+		break;
+	case 1: // sine^2
+		w = std::abs(2.f * std::sin(.5f * p)) - 1.f;
+		break;
+	case 2: // triangle
+		w = p * 2.f * numbers::inv_pi_v<float>;
+		if (w > 1.f) { w = 2.f - w; }
```

Comment:
```suggestion
		w = std::abs((2 / numbers::inv_pi_v<float>) * std::remainder(ph, 2 * numbers::inv_pi_v<float>)) - 1;
```
I found this function which works.

Reply:
![image](https://github.com/user-attachments/assets/32de90c7-a33d-4ca7-9167-f656a095a80e)

Reply:
`std::remainder` is not the same as Desmos's `mod`. In Desmos it would be `x - round(x / y) * y`.

When you do that, you'll get a triangle wave. However, after testing it some more, both your triangle wave and mine appear to be off from the original. I'll look into it some more and see what I can find out.

Reply:
Updated graph; thanks for pointing out the difference between `mod()` and `std::remainder()`.

![image](https://github.com/user-attachments/assets/ac3f4437-840e-4d80-9264-2e51ecf06e7f)

Reply:
Looks like I made a copy-paste mistake by putting `inv_pi_v` here instead of `pi_v`. I've corrected it.
![image](https://github.com/user-attachments/assets/59063a6d-30ce-4d78-83f6-a6fd07b123c9)

Reply:
With the correction to the original triangle wave (the "-1" mistakenly being inside one of the expressions), both your version and mine work according to my testing program:
https://gist.github.com/messmerd/5ee2e0078b574b9878bdd7224192531e